hub/policy/src/main/java/uk/gov/ida/hub/policy/metrics/EidasConnectorMetrics.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.ida.hub.policy.metrics;
+
+import io.prometheus.client.Counter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.Objects;
+
+public class EidasConnectorMetrics {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EidasConnectorMetrics.class);
+
+    public enum Direction {request, response}
+
+    public enum Status {ok, ko, error}
+
+    private static final Counter CONNECTOR_COUNTER = Counter.build(
+            "verify_eidas_connector_count",
+            "EIDAS Connector Journey counter with labels 'country', 'direction', 'status'")
+            .labelNames("country", "direction", "status")
+            .register();
+
+    public static void increment(String entityId, Direction direction, Status status) {
+        try {
+            String country = getCountryCode(Objects.requireNonNull(entityId));
+            CONNECTOR_COUNTER.labels(
+                    country.toLowerCase(),
+                    Objects.requireNonNull(direction.name()),
+                    Objects.requireNonNull(status.name()))
+                    .inc();
+        } catch (Throwable e) {
```

Comment:
Is it standard to do this? If we think it's rare/not expected to ever happen, then it feels weird to turn it into a warning. Particularly _any_ throwable - doesn't Prometheus document what their library may throw?

Reply:
The intent is to ensure that there's no way the `increment` method can break a user journey.
The Prometheus object is safe, the catching here is to deal with Objects.requireNonNull calls, and anything the `getCountryCode` method might throw.

The two exceptions that could be thrown by these methods are IllegalArgumentException & NullPointerException, so I've pushed a small change to narrow our expectation on the type of exception thrown